dvc/utils/compat.py

```diff
@@ -207,6 +207,12 @@ def fspath(path):
         except AttributeError:
             if hasattr(path_type, "__fspath__"):
                 raise
+
+            # Support for pathlib.Path in Python 3.5, since it doesn't
+            # implement `__fspath__`
+            if isinstance(path, pathlib.Path):
+                return str(path)
+
```

Comment:
Why not just use fspath_py35 as we do in all other places?

Reply:
`fspath_py35` doesn't work with `pathlib.Path` in Python 3.5 since the abstract method `__fspath__` was introduced until Python 3.6.

Reply:
and what if we use `pathlib2` for py35 instead of `pathlib`?

Reply:
cross-quoting:

> So maybe we shouldn't use builtin pathlib in Python 3.5 and use pathlib2 instead? but then we will need to adjust PathInfo implementation, some pathes we now do for Python 2 are actually needed to patch diffrences between pathlib and pathlib2.

Reply:
@mroutis Ok, makes sense 🙂 Thanks!

Reply:
sure, @efiop! do you think the comment could be improved?

Reply:
@mroutis Seems alright to me.